hack/e2e-test-ocp.sh

```diff
@@ -94,20 +94,44 @@ function collect_logs {
     $OC describe pods -n kueue-system > "$ARTIFACTS/kueue-system-pods.log" || true
     $OC logs -n kueue-system -l app=kueue --tail=-1 > "$ARTIFACTS/kueue-system-logs.log" || true
     restore_ocp_manager_image
+    restore_kueue_namespace
 }
 
 function deploy_kueue {
-    # Update kueue image
-    (cd config/components/manager-ocp && $KUSTOMIZE edit set image controller="$IMAGE_TAG")
+    local namespace=${KUEUE_NAMESPACE:-kueue-system}
+    (cd config/components/manager-ocp && $KUSTOMIZE edit set image controller="$IMAGE_TAG" && \
+     $KUSTOMIZE edit set namespace "$namespace")
     
     # Deploy kueue
-    $OC apply --server-side -k config/default-ocp
+    local kustomize_path namespace
+    kustomize_path="config/default-ocp"
+
+    # Set namespace in the default-ocp Kustomization
+    (cd "${kustomize_path}" && \
+        $KUSTOMIZE edit set namespace "$namespace")
+
+    namespace="${KUEUE_NAMESPACE:-}"
+    
+    if [[ -n "$namespace" ]]; then
+        # Create a namespace if it doesn't exist
+        $OC create namespace "$namespace" --dry-run=client -o yaml | $OC apply -f -
+        
+        # Apply resources with namespace override
+        $KUSTOMIZE build "${kustomize_path}" | $OC apply --server-side -f -
```

Comment:
I am thinking on this. Feels like the operator should have this functionality.

Reply:
I'm not sure if I understood this, but ideally, this patch is required for upstream submission, which I plan to do later. The goal is to enable running e2e tests in a custom namespace. Once merged, I'll create an openshift/release PR that uses the customized e2e test code, with the kueue operator being installed via the make target we already have in the openshift/kueue-operator.

Reply:
Ah I see. Thanks for clarifying that.

/lgtm
/hold cancel